### PR TITLE
Fix RP2040 compile warning

### PR DESCRIPTION
**Pull request by ActoryOu**
Fix Pico compile warning

Description
-----------

Fixed more warnings mentioned by #731.

Brief:
```sh
In file included from ../3rdParty/FreeRTOS-Kernel/include/FreeRTOS.h:64,
                 from ../3rdParty/FreeRTOS-Kernel/portable/ThirdParty/GCC/RP2040/port.c:34:
../3rdParty/FreeRTOS-Kernel/portable/ThirdParty/GCC/RP2040/port.c: In function 'vYieldCore':
../3rdParty/FreeRTOS-Kernel/portable/ThirdParty/GCC/RP2040/port.c:533:26: warning: comparison of integer expressions of different signedness: 'int' and 'uint' {aka 'unsigned int'} [-Wsign-compare]
  533 |     configASSERT(xCoreID != portGET_CORE_ID());
                                                   ^
[113/114] Building C object CMakeFiles/JadeDrive.dir/3rdParty/FreeRTOS-Kernel/tasks.c.obj
../3rdParty/FreeRTOS-Kernel/tasks.c: In function 'vTaskGetInfo':
../3rdParty/FreeRTOS-Kernel/tasks.c:5549:40: warning: assignment discards 'volatile' qualifier from pointer target type [-Wdiscarded-qualifiers]
 5549 |             pxTaskStatus->pxTopOfStack = pxTCB->pxTopOfStack;
```

Test Steps
-----------

Add below compile options and pass the compilation.
```
target_compile_options( ${TARGET} PUBLIC
    ### Gnu/Clang C Options
    $<$<COMPILE_LANG_AND_ID:C,GNU>:-fdiagnostics-color=always>
    $<$<COMPILE_LANG_AND_ID:C,Clang>:-fcolor-diagnostics>

    $<$<COMPILE_LANG_AND_ID:C,Clang,GNU>:-Wall>
    $<$<COMPILE_LANG_AND_ID:C,Clang,GNU>:-Wextra>
    $<$<COMPILE_LANG_AND_ID:C,Clang,GNU>:-Werror>
    $<$<COMPILE_LANG_AND_ID:C,Clang>:-Weverything>
)
```

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by pranjalchanda08** [comment]
Where are the compilation commands added? I dont see any change in the files w.r.t that.

**Comment by ActoryOu** [comment]
> Where are the compilation commands added? I don't see any change in the files w.r.t that.

I'm using [this repository](https://github.com/ActoryOu/FreeRTOS-Community-Supported-Demos/commit/b2c3b2a50f405fe9ff7c67154d1bbcfff69549bf) to add the compile options. And I need to integrate it into FreeRTOS repo as [CI flow](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/.github/workflows/kernel-demos.yml#L188-L194).

File: portable/ThirdParty/GCC/RP2040/port.c [hunk 1]
```diff
@@ -530,7 +530,11 @@ void vClearInterruptMaskFromISR( __attribute__( ( unused ) ) uint32_t ulMask )
 
 void vYieldCore( int xCoreID )
 {
-    configASSERT(xCoreID != portGET_CORE_ID());
+    /* Remove warning about unused parameters. */
+    ( void ) xCoreID;
```
Comment (chinglee-iot) [inline portable/ThirdParty/GCC/RP2040/port.c:534]: Suggest to make use of this macro here.
configASSERT_DEFINED

Comment (ActoryOu) [reply portable/ThirdParty/GCC/RP2040/port.c:535]: Updated. Because it doesn't impact performance, I updated the comment to describe the reason.

**Review by chinglee-iot — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**